File: ulyp-ui/src/main/java/com/ulyp/ui/code/SourceCodeView.java
package com.ulyp.ui.code;

import javafx.scene.control.TabPane;
import org.springframework.stereotype.Component;

@Component
public class SourceCodeView extends TabPane {

    private final SourceCodeTab mainTab;

    public SourceCodeView() {

        this.mainTab = new SourceCodeTab();
        this.getTabs().add(mainTab);
    }

    public void setText(SourceCode code, String methodNameToScrollTo) {
        this.mainTab.setText(code, methodNameToScrollTo);
    }
}

File: ulyp-ui/src/main/kotlin/com/ulyp/ui/code/SourceCodeView.kt
package com.ulyp.ui.code

import javafx.scene.control.TabPane
import org.springframework.stereotype.Component

@Component
class SourceCodeView(private val mainTab: SourceCodeTab) : TabPane() {

    fun setText(code: SourceCode?, methodNameToScrollTo: String?) {
        mainTab.setText(code!!, methodNameToScrollTo)
    }

    init {
        tabs.add(mainTab)
    }
}